data-prepper-plugins/grok-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepperTests.java

```diff
@@ -11,35 +11,94 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.krakens.grok.api.Grok;
+import io.krakens.grok.api.GrokCompiler;
+import io.krakens.grok.api.Match;
+import com.amazon.dataprepper.model.record.Record;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.MockedStatic;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.Mock;
+import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
 
 @ExtendWith(MockitoExtension.class)
 public class GrokPrepperTests {
 
-    PluginSetting pluginSetting;
+    private PluginSetting pluginSetting;
     private GrokPrepper grokPrepper;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<Map<String, Object>>() {};
+
+    @Mock
+    private GrokCompiler grokCompiler;
+
+    @Mock
+    private Match match;
+
+    @Mock
+    private Grok grok;
+
+    private final String PLUGIN_NAME = "grok";
+    private Map<String, Object> capture;
 
     @BeforeEach
     public void setup() {
-        pluginSetting = new PluginSetting(
-                "grok",
-                null
-                );
-        pluginSetting.setPipelineName("grokPipeline");
-        grokPrepper = new GrokPrepper(pluginSetting);
+
+        try (MockedStatic<GrokCompiler> grokCompilerMockedStatic = mockStatic(GrokCompiler.class)) {
+
+            capture = new HashMap<>();
+            capture.put("field_capture_1", "value_capture_1");
+            capture.put("field_capture_2", "value_capture_2");
+            capture.put("field_capture_3", "value_capture_3");
+
+            final Map<String, List<String>> matchConfig = new HashMap<>();
+            matchConfig.put("message", new ArrayList<String>() {{
+                add("%{COMMONAPACHELOG");
+            }});
+
+            grokCompilerMockedStatic.when(GrokCompiler::newInstance).thenReturn(grokCompiler);
+            when(grokCompiler.compile(eq(matchConfig.get("message").get(0)), anyBoolean())).thenReturn(grok);
+            when(grok.match(anyString())).thenReturn(match);
```

Comment:
Since using random strings doesn't really apply to these tests (because of the merging tests with different types and even some List values, do you propose setting the `messageInput` necessary for each test, and then for each test repeating the `when(grok.match(messageInput)).thenReturn(match);`

Reply:
> do you propose setting the messageInput necessary for each test, and then for each test repeating the when(grok.match(messageInput)).thenReturn(match);

I'm fine with this. But, at the same time, I don't know if we really need that since the `messageInput` is the same in all tests.

What you might need instead is this:

```
void setup() {
  ...
  String messageInput = UUID.randomUUID().toString();
  when(grok.match(messageInput)).thenReturn(match);
  ...
}

// inside some test(s)
String testData = "{\"message\":\"" + messageInput + "\"}";
// inside some other test(s)
String testData = "{\"message\":\"" + messageInput + "\"," +
  "\"field_capture_1\":\"value_capture_collision\"}"
//

```

Reply:
Ah I see what you mean now. I will make this cleaner.